src/main/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterFilter.java

```diff
@@ -248,7 +248,8 @@ public boolean incrementToken() throws IOException {
         // word of no delimiters, or protected word: just return it
         if ((iterator.current == 0 && iterator.end == termLength) ||
                 (protWords != null && protWords.contains(termBuffer, 0, termLength))
-                || protectedWords.contains(termAttribute.toString())) {
+                || protectedWords.contains(termAttribute.toString())
+                || termAttribute.length() == 2) {
```

Comment:
I suggest putting the length check first for performance reasons.

Reply:
I agree, thanks!